File: appkit-creating-and-customizing-the-touch-bar/Objective-C/NSTouchBar Catalog/BackgroundWindow/TitlebarAccessoryViewController.m
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
View controller for the title bar accessory that contains the Set Background button.
*/

#import "TitleBarAccessoryViewController.h"
#import "BackgroundImagesViewController.h"

@implementation TitleBarAccessoryViewController

- (IBAction)presentPhotos:(id)sender
{
    if (self.openingViewController == nil)
    {
        _openingViewController =
            [self.storyboard instantiateControllerWithIdentifier: @"BackgroundImagesViewController"];
    }
    
    [self presentViewController:self.openingViewController
        asPopoverRelativeToRect:self.view.bounds
                         ofView:self.view
                  preferredEdge:NSRectEdgeMinY
                       behavior:NSPopoverBehaviorTransient];
}

@end

File: appkit-creating-and-customizing-the-touch-bar/Swift/NSTouchBar Catalog/TitleBarAccessoryViewController.swift
/*
See the LICENSE.txt file for this sample’s licensing information.

Abstract:
View controller for the title bar accessory that contains the Set Background button.
*/

import Cocoa

class TitleBarAccessoryViewController: NSTitlebarAccessoryViewController {
    var openingViewController: BackgroundImagesViewController!

    @IBAction func presentPhotos(sender: Any) {
        if openingViewController == nil {
            if let imagesViewController =
                self.storyboard?.instantiateController(
                    withIdentifier: NSStoryboard.Name("BackgroundImagesViewController")) as? BackgroundImagesViewController {
                openingViewController = imagesViewController
            }
        }
        // Show the BackgroundImagesViewController as a popover.
        present(openingViewController, asPopoverRelativeTo: self.view.bounds, of: view, preferredEdge: .minY, behavior: .transient)
    }

}
